iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -72,9 +72,24 @@ static bool supportsTensorCore(FuncOp entryPoint, linalg::LinalgOp op) {
   // Limit tensor core pipeline to matmul as not all combinations of transpose
   // are supported upstream.
   // TODO(thomasraoux): Enable batchMatmul and generic contraction.
-  if (getTargetArch(entryPoint) != "sm_80" ||
-      !(isa<linalg::MatmulOp>(op) || isa<linalg::BatchMatmulOp>(op))) {
-    return false;
+  if (getTargetArch(entryPoint) != "sm_80") return false;
+  if (!(isa<linalg::MatmulOp>(op) || isa<linalg::BatchMatmulOp>(op))) {
+    assert(linalg::isaContractionOpInterface(op));
+    // If this is not a named op matmul check some properties to make sure that
+    // we can map it to tensorcore ops. We should have only mulAdd in the region
+    // and the output map should have no permutation and the last dimension
+    // should be a reduce.
+    Region &body = op->getRegion(0);
+    Region::OpIterator it = body.op_begin();
+    if (it == body.op_end() || !isa<arith::MulFOp>(*(it++))) return false;
```

Comment:
ok, I wont pick on this further, but this is still kind of ad-hoc. Its `add(mul(bbArg0, bbArg1), bbArg2))`. Should be written as a `matchPattern` so it can be robust. Your call on whether you want to fix this.
Concern I have here is that this can regress on some totally unrelated change (cant give you off the top of my head what that change might be). Then either it adds tax to someone else cause the CI goes red (or worse if performance regresses), or it will come to you after the fact if we still dont have it tracking as a pre-merge check. Ideally the `ContractionOpInterface` already does this chaeck for you, but that interface is broken in so many ways....

Reply:
Yes the idea is that I didn't want to redo the work of the ContractionOpInterface. I do think we need better matcher, I was discussing with Nicolas and there is some new mechanism in iree-dialect but it didn't seem to be the right time to bring that here. This is definitely hugely but it is safe. I do think we want to support better matcher to handle transposed matmul etc... but I didn't want to start adding a lot of code since there doesn't seem to be a consensus on how it should look like at this point.